Queue/Backend/Redis.php

```diff
@@ -118,6 +118,9 @@ public function getFirstXValuesFromList($key, $numValues)
 
         $this->connectIfNeeded();
         $values = $this->redis->lRange($key, 0, $numValues - 1);
+        foreach($values as $key => $value) {
+            $values[$key] = gzinflate($value);
```

Comment:
you will need to check if it returns a non empty result and if not, use `$value`. Otherwise if you update the plugin, and still have eg 1M requests queued, those would all be `null`.

Reply:
Didn't understand that. `foreach` only happens if there is data returned, if not it will return $value as is, empty(?)

Reply:
`gzinflate($value);` may return an empty result if the content is not deflated. Which all entries are that were queued before the plugin update